### ISSUE TITLE
how to extract feature on pooling layer

### ISSUE THREAD

**Issue by ForestWang** [davidsandberg/facenet#363]
in the mode of inception_resnet_v1, I want to extract feature after pooling layers in stead of FC layers, 
"net = slim.avg_pool2d(net, net.get_shape()[1:3], padding='VALID', scope='AvgPool_1a_8x8')".

I;m a beginner with tensor flow, is anyone can help me how to write the codes with python. thanks

**Comment by AKSHAYUBHAT** [comment]
you can get a specific tensor by name like on this line. 

https://github.com/AKSHAYUBHAT/DeepVideoAnalytics/blob/master/dvalib/indexer.py#L264

You can load the graph and print names of tensors to select the one you want.

https://github.com/AKSHAYUBHAT/VisualSearchServer/blob/master/notebooks/notebook_network.ipynb

**Comment by ForestWang** [comment]
Thank you very much.

 I use your method print out all tensor name, but not found the pooling layers name. the model is '20170512-110547'

Operation: save/Assign_489
save/Assign_489 Input  InceptionResnetV1/Repeat_2/block8_5/Conv2d_1x1/weights:0 (1, 1, 384, 1792)
save/Assign_489 Input  save/RestoreV2_489:0 <unknown>
save/Assign_489 Output  save/Assign_489:0 (1, 1, 384, 1792)


Operation: save/RestoreV2_490/tensor_names
save/RestoreV2_490/tensor_names Output  save/RestoreV2_490/tensor_names:0 (1,)


Operation: save/RestoreV2_490/shape_and_slices
save/RestoreV2_490/shape_and_slices Output  save/RestoreV2_490/shape_and_slices:0 (1,)


Operation: save/RestoreV2_490
save/RestoreV2_490 Input  save/Const:0 ()
save/RestoreV2_490 Input  save/RestoreV2_490/tensor_names:0 (1,)
save/RestoreV2_490 Input  save/RestoreV2_490/shape_and_slices:0 (1,)
save/RestoreV2_490 Output  save/RestoreV2_490:0 <unknown>


Operation: save/Assign_490
save/Assign_490 Input  Logits/biases:0 (44052,)
save/Assign_490 Input  save/RestoreV2_490:0 <unknown>
save/Assign_490 Output  save/Assign_490:0 (44052,)


Operation: save/RestoreV2_491/tensor_names
save/RestoreV2_491/tensor_names Output  save/RestoreV2_491/tensor_names:0 (1,)


Operation: save/RestoreV2_491/shape_and_slices
save/RestoreV2_491/shape_and_slices Output  save/RestoreV2_491/shape_and_slices:0 (1,)


Operation: save/RestoreV2_491
save/RestoreV2_491 Input  save/Const:0 ()
save/RestoreV2_491 Input  save/RestoreV2_491/tensor_names:0 (1,)
save/RestoreV2_491 Input  save/RestoreV2_491/shape_and_slices:0 (1,)
save/RestoreV2_491 Output  save/RestoreV2_491:0 <unknown>


Operation: save/Assign_491
save/Assign_491 Input  Logits/weights:0 (128, 44052)
save/Assign_491 Input  save/RestoreV2_491:0 <unknown>
save/Assign_491 Output  save/Assign_491:0 (128, 44052)


Operation: save/restore_all


Operation: Merge/MergeSummary
Merge/MergeSummary Input  input_producer:0 ()
Merge/MergeSummary Input  batch_join/fraction_of_1440_full:0 ()
Merge/MergeSummary Input  learning_rate_1:0 ()
Merge/MergeSummary Input  cross_entropy__raw_:0 ()
Merge/MergeSummary Input  cross_entropy_1:0 ()
Merge/MergeSummary Input  total_loss__raw_:0 ()
Merge/MergeSummary Input  total_loss_1:0 ()
Merge/MergeSummary Output  Merge/MergeSummary:0 ()


Operation: init


Operation: init_1
